src/App.tsx

```diff
@@ -28,11 +29,27 @@ type ReorderOptions = {
 
 // Use this function in the render method to prepare goods
 export function getReorderedGoods(
-  goods: string[],
   { sortType, isReversed }: ReorderOptions,
 ) {
-  // To avoid the original array mutation
-  const visibleGoods = [...goods];
+  let visibleGoods = [...goodsFromServer];
+
+  if (sortType === SortType.ALPHABET) {
+    visibleGoods = visibleGoods.sort((a, b) => a.localeCompare(b));
+  } else if (sortType === SortType.LENGTH) {
+    visibleGoods = visibleGoods.sort((a, b) => {
+      const lengthDiffrence = a.length - b.length;
+
+      if (lengthDiffrence === 0) {
+        return a.localeCompare(b);
+      }
+
+      return lengthDiffrence;
+    });
+  }
```

Comment:
You don't need to use `let visibleGoods` and then later `visibleGoods = visibleGoods.sort(...)`, with `const visibleGoods` should work just with ` visibleGoods.sort(...)`, because `.sort()`, `reverse()`, and couple more operations are mutating the original array, here:  `const visibleGoods`  and will change how the array looks :)